lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,472 @@
+    #[clap(long, name = "no-cargo-wapm")]
+    pub no_cargo_wapm: bool,
+    /// Namespace to init with, default = current logged in user or _
+    #[clap(long, name = "namespace")]
+    pub namespace: Option<String>,
+    /// Version of the initialized package
+    #[clap(long, name = "version")]
+    pub version: Option<semver::Version>,
+    /// If the `manifest-path` is a Cargo.toml, use that file to initialize the wapm.toml
+    #[clap(long, name = "manifest-path")]
+    pub manifest_path: Option<PathBuf>,
+    /// Add default dependencies for common packages (currently supported: `python`, `js`)
+    #[clap(long, name = "template")]
+    pub template: Option<String>,
+    /// Include file paths into the target container filesystem
+    #[clap(long, name = "include")]
+    pub include: Vec<String>,
+    /// Directory of the output file name. wasmer init will error in the target dir
+    /// already contains a wasmer.toml. Also sets the package name.
+    #[clap(name = "PACKAGE_PATH")]
+    pub out: Option<PathBuf>,
+}
+
+#[derive(Debug, PartialEq, Copy, Clone)]
+enum BinOrLib {
+    Bin,
+    Lib,
+    Empty,
+}
+
+// minimal version of the Cargo.toml [package] section
+#[derive(Debug, Clone)]
+struct MiniCargoTomlPackage {
+    name: String,
+    version: semver::Version,
+    description: Option<String>,
+    homepage: Option<String>,
+    repository: Option<String>,
+    license: Option<String>,
+    readme: Option<PathBuf>,
+    license_file: Option<PathBuf>,
+    #[allow(dead_code)]
+    workspace_root: PathBuf,
+    #[allow(dead_code)]
+    build_dir: PathBuf,
+}
+
+static WASMER_TOML_NAME: &str = "wasmer.toml";
+
+impl Init {
+    /// `wasmer init` execution
+    pub fn execute(&self) -> Result<(), anyhow::Error> {
+        let bin_or_lib = self.get_bin_or_lib()?;
+
+        let package_name;
+        let target_file = match self.out.as_ref() {
+            None => {
+                let current_dir = std::env::current_dir()?;
+                package_name = current_dir
+                    .canonicalize()?
+                    .file_stem()
+                    .and_then(|s| s.to_str())
+                    .map(|s| s.to_string())
+                    .ok_or_else(|| anyhow::anyhow!("no current dir name"))?;
+                current_dir.join(WASMER_TOML_NAME)
+            }
+            Some(s) => {
+                let _ = std::fs::create_dir_all(s);
+                package_name = s
+                    .canonicalize()?
+                    .file_stem()
+                    .and_then(|s| s.to_str())
+                    .map(|s| s.to_string())
+                    .ok_or_else(|| anyhow::anyhow!("no dir name"))?;
+                s.join(WASMER_TOML_NAME)
+            }
+        };
+
+        if target_file.exists() && !self.overwrite {
+            return Err(anyhow::anyhow!(
+                "wapm project already initialized in {}",
+                target_file.display()
+            ));
```

Comment:
It doesn't help much here because the message's length and current indentation level mean the return is spread over multiple lines, but sometimes `anyhow::bail!()` is more concise.

```suggestion
            anyhow::bail!(
                "wapm project already initialized in {}",
                target_file.display(),
            );
```